projects/control-service/projects/pipelines_control_service/src/main/java/com/vmware/taurus/service/deploy/DeploymentProgressV2.java

```diff
@@ -0,0 +1,87 @@
+/*
+ * Copyright 2021-2023 VMware, Inc.
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package com.vmware.taurus.service.deploy;
+
+import com.vmware.taurus.service.diag.methodintercept.Measurable;
+import com.vmware.taurus.service.model.*;
+import com.vmware.taurus.service.monitoring.DeploymentMonitorV2;
+import com.vmware.taurus.service.notification.DataJobNotification;
+import lombok.RequiredArgsConstructor;
+import org.springframework.stereotype.Component;
+
+/** Track the progress of a deployment. */
+@Component
+@RequiredArgsConstructor
+public class DeploymentProgressV2 {
```

Comment:
In hindsight, V2 are really hard to review. What are the differences?

If there are not significant (especially breaking) difference, might be better to extend V1

Reply:
Extended the most of V1 classes.